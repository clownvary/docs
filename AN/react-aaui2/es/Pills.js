import React from 'react';
import { string, object, node, oneOf } from 'prop-types';
import classNames from 'classnames';

var Pills = function Pills(_ref) {
  var _classNames;

  var className = _ref.className,
      style = _ref.style,
      size = _ref.size,
      children = _ref.children;

  var classes = classNames((_classNames = {
    'btn-group': true
  }, _classNames['btn-group-' + size] = size, _classNames), className);

  return React.createElement(
    'div',
    { className: classes, style: style },
    children
  );
};

Pills.displayName = 'AUIPills';
Pills.propTypes = {
  className: string,
  style: object, // eslint-disable-line
  size: oneOf(['xs', 'sm', 'lg', 'xl']),
  children: node
};

export default Pills;