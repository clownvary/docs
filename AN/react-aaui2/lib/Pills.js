'use strict';

exports.__esModule = true;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Pills = function Pills(_ref) {
  var _classNames;

  var className = _ref.className,
      style = _ref.style,
      size = _ref.size,
      children = _ref.children;

  var classes = (0, _classnames2.default)((_classNames = {
    'btn-group': true
  }, _classNames['btn-group-' + size] = size, _classNames), className);

  return _react2.default.createElement(
    'div',
    { className: classes, style: style },
    children
  );
};

Pills.displayName = 'AUIPills';
Pills.propTypes = {
  className: _propTypes.string,
  style: _propTypes.object, // eslint-disable-line
  size: (0, _propTypes.oneOf)(['xs', 'sm', 'lg', 'xl']),
  children: _propTypes.node
};

exports.default = Pills;
module.exports = exports['default'];