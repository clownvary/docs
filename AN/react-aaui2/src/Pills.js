import React from 'react'
import { string, object, node, oneOf } from 'prop-types'
import classNames from 'classnames'

const Pills = ({ className, style, size, children }) => {
  const classes = classNames(
    {
      'btn-group': true,
      [`btn-group-${size}`]: size,
    },
    className,
  )

  return (
    <div className={classes} style={style}>
      {children}
    </div>
  )
}

Pills.displayName = 'AUIPills'
Pills.propTypes = {
  className: string,
  style: object, // eslint-disable-line
  size: oneOf(['xs', 'sm', 'lg', 'xl']),
  children: node,
}

export default Pills
