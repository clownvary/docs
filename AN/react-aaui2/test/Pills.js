import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import Pills from 'Pills'

it('should render without errors', () => {
  const wrapper = shallow(<Pills />)

  expect(wrapper).toBeTruthy()
})

it('should render right thing', () => {
  const wrapper = shallow(<Pills />)

  expect(wrapper.hasClass('btn-group')).toBe(true)
})

it("should render right thing and don't change unexpected", () => {
  const tree = renderer.create(<Pills />).toJSON()

  expect(tree).toMatchSnapshot()
})

it("should render right thing and don't change unexpected with different sizes", () => {
  const sizes = ['xs', 'sm', 'lg', 'xl']

  sizes.forEach(size => {
    const tree = renderer.create(<Pills size={size} />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})

it("should accept different sizes(['xs', 'sm', 'lg', 'xl'])", () => {
  const smPills = shallow(<Pills size="sm" />)

  expect(smPills.hasClass('btn-group-sm')).toBe(true)
})
