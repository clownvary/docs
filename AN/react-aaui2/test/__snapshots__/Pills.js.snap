// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render right thing and don't change unexpected 1`] = `
<div
  className="btn-group"
  style={undefined}
/>
`;

exports[`should render right thing and don't change unexpected with different sizes 1`] = `
<div
  className="btn-group btn-group-xs"
  style={undefined}
/>
`;

exports[`should render right thing and don't change unexpected with different sizes 2`] = `
<div
  className="btn-group btn-group-sm"
  style={undefined}
/>
`;

exports[`should render right thing and don't change unexpected with different sizes 3`] = `
<div
  className="btn-group btn-group-lg"
  style={undefined}
/>
`;

exports[`should render right thing and don't change unexpected with different sizes 4`] = `
<div
  className="btn-group btn-group-xl"
  style={undefined}
/>
`;
